### ISSUE TITLE
[BUG] DirectReductionForecaster fails with DataFrame input

### ISSUE THREAD

**Issue by aiwalter** [sktime/sktime#3491]
**Describe the bug**
```python
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_27752\3920336442.py in <module>
     27     # error_score='raise',
     28 )
---> 29 gscv.fit(y, fh=fh)
     30 y_pred = gscv.predict(fh=fh)

c:\Users\mf-wa\Desktop\sktime\sktime\sktime\forecasting\base\_base.py in fit(self, y, X, fh)
    301         # we call the ordinary _fit if no looping/vectorization needed
    302         if not vectorization_needed:
--> 303             self._fit(y=y_inner, X=X_inner, fh=fh)
    304         else:
    305             # otherwise we call the vectorized version of fit

c:\Users\mf-wa\Desktop\sktime\sktime\sktime\forecasting\model_selection\_tune.py in _fit(self, y, X, fh)
    183 
    184         # Run grid-search cross-validation.
--> 185         results = self._run_search(evaluate_candidates)
    186 
    187         results = pd.DataFrame(results)

c:\Users\mf-wa\Desktop\sktime\sktime\sktime\forecasting\model_selection\_tune.py in _run_search(self, evaluate_candidates)
    477         """Search all candidates in param_grid."""
    478         self._check_param_grid(self.param_grid)
--> 479         return evaluate_candidates(ParameterGrid(self.param_grid))
    480 
    481     @classmethod

c:\Users\mf-wa\Desktop\sktime\sktime\sktime\forecasting\model_selection\_tune.py in evaluate_candidates(candidate_params)
    170 
    171             out = parallel(
--> 172                 delayed(_fit_and_score)(params) for params in candidate_params
    173             )
    174 

c:\Users\mf-wa\miniconda3\envs\test37\lib\site-packages\joblib\parallel.py in __call__(self, iterable)
   1096 
   1097             with self._backend.retrieval_context():
-> 1098                 self.retrieve()
   1099             # Make sure that we get a last message telling us we are done
   1100             elapsed_time = time.time() - self._start_time

c:\Users\mf-wa\miniconda3\envs\test37\lib\site-packages\joblib\parallel.py in retrieve(self)
    973             try:
    974                 if getattr(self._backend, 'supports_timeout', False):
--> 975                     self._output.extend(job.get(timeout=self.timeout))
    976                 else:
    977                     self._output.extend(job.get())

c:\Users\mf-wa\miniconda3\envs\test37\lib\site-packages\joblib\_parallel_backends.py in wrap_future_result(future, timeout)
    565         AsyncResults.get from multiprocessing."""
    566         try:
--> 567             return future.result(timeout=timeout)
    568         except CfTimeoutError as e:
...
--> 384             raise self._exception
    385         else:
    386             return self._result

NotImplementedError: <class 'pandas.core.indexes.multi.MultiIndex'> is not supported for input, use one of (<class 'pandas.core.indexes.range.RangeIndex'>, <class 'pandas.core.indexes.period.PeriodIndex'>, <class 'pandas.core.indexes.datetimes.DatetimeIndex'>, <class 'pandas.core.indexes.timedeltas.TimedeltaIndex'>) instead.
```
**To Reproduce**

Fails:
```python
from sktime.forecasting.compose._reduce import DirectReductionForecaster
from sktime.datatypes import get_examples
from sktime.forecasting.model_selection import ExpandingWindowSplitter
from sktime.forecasting.model_selection import ForecastingGridSearchCV
from sklearn.ensemble import RandomForestRegressor

y = get_examples(mtype="pd_multiindex_hier")[1]
forecaster = DirectReductionForecaster(
    estimator=RandomForestRegressor(random_state=1),
    window_length=3,
    pooling="global",
)
fh=[1,2,3]
cv = ExpandingWindowSplitter(
    initial_window=3,
    step_length=3,
    start_with_window=True,
    fh=fh,
)
gscv = ForecastingGridSearchCV(
    forecaster=forecaster,
    param_grid={
        "estimator__n_estimators": [50, 70]
    },
    cv=cv,
    n_jobs=-1,
    # error_score='raise',
)
gscv.fit(y, fh=fh)
y_pred = gscv.predict(fh=fh)
```
If I change the fitting line to `gscv.fit(y.values, fh=fh)` it works. However some fits have failed.

**Additional context**

**Versions**
main

**Comment by fkiraly** [comment]
ah, that's because grid search evaluate does not work with hierarchical data properly, I think

reason is likely the same as this issue:
https://github.com/alan-turing-institute/sktime/issues/3464

**Comment by aiwalter** [comment]
so i guess the `numpy` input was also not processed correctly, but for some reason it was passing

**Comment by fkiraly** [comment]
The underlying bug (in grid search) has been fixed in 0.14.0, but the code is now failing due to the data not being large enough for the splitter (data has length 3, but splitter requires >5).

The example with longer data works now:

```python
from sktime.forecasting.compose._reduce import DirectReductionForecaster
from sktime.forecasting.model_selection import ExpandingWindowSplitter
from sktime.forecasting.model_selection import ForecastingGridSearchCV
from sklearn.ensemble import RandomForestRegressor
from sktime.utils._testing.panel import _make_panel

y = _make_panel()
forecaster = DirectReductionForecaster(
    estimator=RandomForestRegressor(random_state=1),
    window_length=3,
    pooling="global",
)
fh=[1,2,3]
cv = ExpandingWindowSplitter(
    initial_window=6,
    step_length=3,
    start_with_window=True,
    fh=fh,
)
gscv = ForecastingGridSearchCV(
    forecaster=forecaster,
    param_grid={
        "estimator__n_estimators": [50, 70]
    },
    cv=cv,
    n_jobs=-1,
    # error_score='raise',
)
gscv.fit(y, fh=fh)
y_pred = gscv.predict(fh=fh)
```
